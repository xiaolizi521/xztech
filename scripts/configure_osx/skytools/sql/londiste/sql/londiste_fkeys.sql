
set log_error_verbosity = 'terse';

create table ref_1 (
    id int4 primary key,
    val text
);

create table ref_2 (
    id int4 primary key,
    ref int4 not null references ref_1,
    val text
);

create table ref_3 (
    id int4 primary key,
    ref2 int4 not null references ref_2,
    val text
);

select * from londiste.subscriber_add_table('refqueue', 'public.ref_1');
select * from londiste.subscriber_add_table('refqueue', 'public.ref_2');
select * from londiste.subscriber_add_table('refqueue', 'public.ref_3');

select * from londiste.find_table_fkeys('public.ref_1');
select * from londiste.find_table_fkeys('public.ref_2');
select * from londiste.find_table_fkeys('public.ref_3');

select * from londiste.subscriber_get_table_pending_fkeys('public.ref_2');

select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');

-- drop fkeys

select * from londiste.subscriber_drop_table_fkey('public.ref_2', 'ref_2_ref_fkey');

select * from londiste.find_table_fkeys('public.ref_1');
select * from londiste.find_table_fkeys('public.ref_2');
select * from londiste.find_table_fkeys('public.ref_3');

select * from londiste.subscriber_drop_table_fkey('public.ref_3', 'ref_3_ref2_fkey');

-- check if dropped

select * from londiste.find_table_fkeys('public.ref_1');
select * from londiste.find_table_fkeys('public.ref_2');
select * from londiste.find_table_fkeys('public.ref_3');

-- look state
select * from londiste.subscriber_get_table_pending_fkeys('public.ref_2');
select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');

-- toggle sync
select * from londiste.subscriber_set_table_state('refqueue', 'public.ref_1', null, 'ok');
select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
select * from londiste.subscriber_set_table_state('refqueue', 'public.ref_2', null, 'ok');
select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
select * from londiste.subscriber_set_table_state('refqueue', 'public.ref_3', null, 'ok');
select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');

-- restore
select * from londiste.subscriber_restore_table_fkey('public.ref_2', 'ref_2_ref_fkey');
select * from londiste.subscriber_restore_table_fkey('public.ref_3', 'ref_3_ref2_fkey');

-- look state
select * from londiste.subscriber_get_table_pending_fkeys('public.ref_2');
select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
select * from londiste.find_table_fkeys('public.ref_1');
select * from londiste.find_table_fkeys('public.ref_2');
select * from londiste.find_table_fkeys('public.ref_3');


