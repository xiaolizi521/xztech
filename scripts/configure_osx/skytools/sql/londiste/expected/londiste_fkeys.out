set log_error_verbosity = 'terse';
create table ref_1 (
    id int4 primary key,
    val text
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "ref_1_pkey" for table "ref_1"
create table ref_2 (
    id int4 primary key,
    ref int4 not null references ref_1,
    val text
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "ref_2_pkey" for table "ref_2"
create table ref_3 (
    id int4 primary key,
    ref2 int4 not null references ref_2,
    val text
);
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "ref_3_pkey" for table "ref_3"
select * from londiste.subscriber_add_table('refqueue', 'public.ref_1');
 subscriber_add_table 
----------------------
                    0
(1 row)

select * from londiste.subscriber_add_table('refqueue', 'public.ref_2');
 subscriber_add_table 
----------------------
                    0
(1 row)

select * from londiste.subscriber_add_table('refqueue', 'public.ref_3');
 subscriber_add_table 
----------------------
                    0
(1 row)

select * from londiste.find_table_fkeys('public.ref_1');
  from_table  |   to_table   |   fkey_name    |                                              fkey_def                                              
--------------+--------------+----------------+----------------------------------------------------------------------------------------------------
 public.ref_2 | public.ref_1 | ref_2_ref_fkey | alter table only public.ref_2 add constraint ref_2_ref_fkey FOREIGN KEY (ref) REFERENCES ref_1(id)
(1 row)

select * from londiste.find_table_fkeys('public.ref_2');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_2 | public.ref_1 | ref_2_ref_fkey  | alter table only public.ref_2 add constraint ref_2_ref_fkey FOREIGN KEY (ref) REFERENCES ref_1(id)
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(2 rows)

select * from londiste.find_table_fkeys('public.ref_3');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(1 row)

select * from londiste.subscriber_get_table_pending_fkeys('public.ref_2');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

-- drop fkeys
select * from londiste.subscriber_drop_table_fkey('public.ref_2', 'ref_2_ref_fkey');
 subscriber_drop_table_fkey 
----------------------------
                          1
(1 row)

select * from londiste.find_table_fkeys('public.ref_1');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

select * from londiste.find_table_fkeys('public.ref_2');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(1 row)

select * from londiste.find_table_fkeys('public.ref_3');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(1 row)

select * from londiste.subscriber_drop_table_fkey('public.ref_3', 'ref_3_ref2_fkey');
 subscriber_drop_table_fkey 
----------------------------
                          1
(1 row)

-- check if dropped
select * from londiste.find_table_fkeys('public.ref_1');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

select * from londiste.find_table_fkeys('public.ref_2');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

select * from londiste.find_table_fkeys('public.ref_3');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

-- look state
select * from londiste.subscriber_get_table_pending_fkeys('public.ref_2');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_2 | public.ref_1 | ref_2_ref_fkey  | alter table only public.ref_2 add constraint ref_2_ref_fkey FOREIGN KEY (ref) REFERENCES ref_1(id)
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(2 rows)

select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

-- toggle sync
select * from londiste.subscriber_set_table_state('refqueue', 'public.ref_1', null, 'ok');
 subscriber_set_table_state 
----------------------------
                          1
(1 row)

select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

select * from londiste.subscriber_set_table_state('refqueue', 'public.ref_2', null, 'ok');
 subscriber_set_table_state 
----------------------------
                          1
(1 row)

select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
  from_table  |   to_table   |   fkey_name    |                                              fkey_def                                              
--------------+--------------+----------------+----------------------------------------------------------------------------------------------------
 public.ref_2 | public.ref_1 | ref_2_ref_fkey | alter table only public.ref_2 add constraint ref_2_ref_fkey FOREIGN KEY (ref) REFERENCES ref_1(id)
(1 row)

select * from londiste.subscriber_set_table_state('refqueue', 'public.ref_3', null, 'ok');
 subscriber_set_table_state 
----------------------------
                          1
(1 row)

select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_2 | public.ref_1 | ref_2_ref_fkey  | alter table only public.ref_2 add constraint ref_2_ref_fkey FOREIGN KEY (ref) REFERENCES ref_1(id)
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(2 rows)

-- restore
select * from londiste.subscriber_restore_table_fkey('public.ref_2', 'ref_2_ref_fkey');
 subscriber_restore_table_fkey 
-------------------------------
                             1
(1 row)

select * from londiste.subscriber_restore_table_fkey('public.ref_3', 'ref_3_ref2_fkey');
 subscriber_restore_table_fkey 
-------------------------------
                             1
(1 row)

-- look state
select * from londiste.subscriber_get_table_pending_fkeys('public.ref_2');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

select * from londiste.subscriber_get_queue_valid_pending_fkeys('refqueue');
 from_table | to_table | fkey_name | fkey_def 
------------+----------+-----------+----------
(0 rows)

select * from londiste.find_table_fkeys('public.ref_1');
  from_table  |   to_table   |   fkey_name    |                                              fkey_def                                              
--------------+--------------+----------------+----------------------------------------------------------------------------------------------------
 public.ref_2 | public.ref_1 | ref_2_ref_fkey | alter table only public.ref_2 add constraint ref_2_ref_fkey FOREIGN KEY (ref) REFERENCES ref_1(id)
(1 row)

select * from londiste.find_table_fkeys('public.ref_2');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_2 | public.ref_1 | ref_2_ref_fkey  | alter table only public.ref_2 add constraint ref_2_ref_fkey FOREIGN KEY (ref) REFERENCES ref_1(id)
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(2 rows)

select * from londiste.find_table_fkeys('public.ref_3');
  from_table  |   to_table   |    fkey_name    |                                               fkey_def                                               
--------------+--------------+-----------------+------------------------------------------------------------------------------------------------------
 public.ref_3 | public.ref_2 | ref_3_ref2_fkey | alter table only public.ref_3 add constraint ref_3_ref2_fkey FOREIGN KEY (ref2) REFERENCES ref_2(id)
(1 row)

